resources/gcloud.rb

```diff
@@ -9,90 +9,284 @@
+    # Installing gcloud autocomplete via apt-get will be faster
+    # issue (https://github.com/teracyhq-incubator/kubernetes-stack-cookbook/issues/19)
+    if platform?('ubuntu')
+      version_avaiable = version_avaiable_in_apt_package(arch_core, version) unless arch_core.to_s.strip.empty?
+    end
+
+    # Gcloud version will be installed via apt-get
+    if version_avaiable == true
+      execute 'import google-cloud-sdk public key' do
+        command 'curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -'
+      end
+
+      apt_repository 'google-cloud-sdk' do
+        uri          'http://packages.cloud.google.com/apt'
+        distribution "cloud-sdk-#{node['lsb']['codename']}"
+        components   ['main']
+        # key 'A7317B0F'
+        # keyserver 'packages.cloud.google.com/apt/doc/apt-key.gpg'
+      end
+
+      if version == latest_version
+        package 'google-cloud-sdk'
+      else
+        package 'google-cloud-sdk' do
+          version "#{version}-0"
+        end
+      end
+
+      # Link to binary_path
+      link_to_binary_path(binary_path)
+      # Disable update notification when run command 'gcloud version'
+      disable_update_check
+    # Gcloud version will be installed via file downloaded
+    else
+      # Check requirement to install
+      install_requirement_for_ubuntu if platform?('ubuntu')
+      install_requirement_for_centos if platform?('centos')
+
+      download_url = "https://storage.googleapis.com/cloud-sdk-release/google-cloud-sdk-#{version}-#{platform}-#{arch}.tar.gz"
+
+      execute "curl #{download_url} | tar xvz" do
+        cwd '/usr/lib'
+        action :run
+        not_if 'which gcloud'
+      end
+
+      execute './google-cloud-sdk/install.sh --quiet' do
+        cwd '/usr/lib'
+        action :run
+        only_if 'test -d /usr/lib/google-cloud-sdk'
+      end
+
+      # Link to binary_path
+      link_to_binary_path(binary_path)
+      # Disable update notification when run command 'gcloud version'
+      disable_update_check
+      # Install autocomplete
+      install_autocomplete
+    end
+  end
+end
+
+action :remove do
+  delete_gcloud(binary_path)
+end
+
+action_class do
+  def attr_platform
+    platform_cmd = Mixlib::ShellOut.new('uname')
+    platform_cmd.run_command
+    platform_cmd.error!
+    platform = platform_cmd.stdout.strip.downcase
+    platform
+  end
+
+  def attr_arch
+    arch_cmd = Mixlib::ShellOut.new('uname -m')
+    arch_cmd.run_command
+    arch_cmd.error!
+    arch = arch_cmd.stdout.strip
+    arch
+  end
+
+  def attr_arch_core(arch)
+    case arch
+    when 'x86', 'i686', 'i386'
+      arch_core = '386'
+    when 'x86_64', 'aarch64'
+      arch_core = 'amd64'
+    when 'armv5*'
+      arch_core = 'armv5'
+    when 'armv6*'
+      arch_core = 'armv6'
+    when 'armv7*'
+      arch_core = 'armv7'
+    end
+    arch_core
```

Comment:
```arch_core = case arch
            when 'x86', 'i686', 'i386'
               '386'
            when 'x86_64', 'aarch64'
              'amd64'
            when 'armv5*'
              'armv5'
            when 'armv6*'
              'armv6'
            when 'armv7*'
              'armv7'
            end```
We can try this.
Also I think we may need a default value for arch_core if arch is none of the option above? (I dont know much about the logic, maybe you had cover all possible cases, just to make sure though...)